src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs

```diff
@@ -823,6 +832,28 @@ private void WriteProjectItems(IEnumerable items)
             }
         }
 
+        private const string EmbedInBinlogItemType = "EmbedInBinlog";
+
+        private void CheckForFilesToEmbed(string itemType, object itemList)
+        {
```

Comment:
Would it make sense to check the `EmbedFile` event for null here and bail right away?

Reply:
In theory, but in our scenario we know it’s not null, so probably doesn’t matter either way. I was just using the ?.Invoke pattern out of habit.

Reply:
Apologies for scrutinizing and hope I'm not misreading the code. `EmbedFile` is subscribed to always, yes, but the handler `EventArgsWriter_EmbedFile` is a conditional no-op. And that condition looks real, i.e. `projectImportsCollector` does not always get created. Are you sure we can't hoist the check and avoid running the loop here in some scenarios?

Reply:
Oh, I see what you're saying. Sounds good.

Reply:
How's this?

Reply:
Awesome, thank you!